### PR TITLE
 Fixes the join problem

### PR DESCRIPTION
**Pull request by SlyDeath**
When SQL error "key" is ambiguous

### CODE REVIEW

**Comment by lazychaser** [comment]
This is incompatible with previous versions of Laravel that doesn't have `getQualifiedKeyName`. If you fix this, I'll merge then

**Comment by SlyDeath** [comment]
Done

File: src/QueryBuilder.php [hunk 1]
```diff
@@ -86,7 +86,7 @@ public function whereIsRoot()
      */
     public function whereAncestorOf($id, $andSelf = false, $boolean = 'and')
     {
-        $keyName = $this->model->getKeyName();
+        $keyName = $this->model->getTable() . '.' . $this->model->getKeyName();
```
Comment (oujesky) [inline src/QueryBuilder.php]: There is a problem here, as `$keyName` is also used below on line 103 in subquery and this needs to be in form of really just the key column name. If the table name is specified here, it won't work as the table is referenced by alias `_` in the subquery. 

`$keyName` should stay as is and the table name should be added only on line 117, where it is the only place where it is actually needed and prevents `Column '...' in where clause is ambiguous .` errors.

Comment (SlyDeath) [inline src/QueryBuilder.php]: Done

**Comment by tortuetorche** [comment]
Hi,

Any news of this pull request?
By the way, this package is amazing 👍 

Cheers,
Tortue Torche

**Pull request merged**